src/git/git.go

```diff
@@ -181,3 +184,122 @@ func CreatedRepo(r *http.Request) (*Repository, int, error) {
 		},
 	}, 200, nil
 }
+
+func validateActionRequest(r *http.Request) bool {
+	headerValue := r.Header.Get("Authorization")
+	return "Bearer " + githubActionToken == headerValue
+}
+
+// CIHandler handles the request to add CI Files to a repo
+func CIHandler(w http.ResponseWriter, r *http.Request) {
+
+	if !validateActionRequest(r) {
+		log.Info("Request verification from Github Action: FAILED")
+		w.WriteHeader(403)
+		return
+	}
+	log.Info("Request verification from Github Action: SUCCESS")
+
+	if r.Method != "POST" {
+		w.WriteHeader(405)
+		log.Errorf("received a %s request, expected POST for CI Handler", r.Method)
+		return
+	}
+
+	r_body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(500)
+		log.Errorf("cannot read request body - %v", err)
+		return
+	}
+
+	var ci CIAction
+	if err := json.Unmarshal(r_body, &ci); err != nil {
+		w.WriteHeader(500)
+		log.Errorf("cannot unmarshal request body - %v", err)
+		return
+	}
+
+	var ciList []string
+	// add Default checks
+	ciList = append(ciList, "Default")
+	v := reflect.ValueOf(ci)
+	for i := 0; i < v.NumField(); i++ {
+		if v.Field(i).String() == "true" {
+			ciList = append(ciList, v.Type().Field(i).Name)
+		}
+	}
+	addChecks(ciList)
+	log.Info("Added checks to pre-commit-config.yaml")
+
+	// call a script to add this file to the repo
+
+	// get ssh_url of the repo
+	repositories, err := Repos()
+	if err != nil {
+		log.Error("Error getting repositories %v", err)
+		return
+	}
+	var repo Repository
+	for _, repository := range repositories {
+		if repository.Name == ci.Repo {
+			repo = repository
+		}
+	}
+	if repo.Name == "" {
+		log.Error("Repository not found")
+		return
+	}
+
+	for _, branch := range repo.Branches {
+		go addCI(repo.URL, repo.Name, branch)
+	}
+
+}
+
+func addChecks(ciList []string) {
+	// combine the data of all files in ciList to a file named .pre-commit-config.yaml
+
+	// create the file even if it exists
+	os.Create(PRE_COMMIT_CONFIG)
```

Comment:
I am not very comfortable with the idea of using the same file `PRE_COMMIT_CONFIG` for every call to `addChecks`. Imagine if there were two consecutive requests for adding CI checks (maybe the user was impatient, and didn't like to wait). Then this file can be overwritten while `ci.sh` is being executed and this can lead to hard-to-spot bugs. I would suggest some alternatives, you are free to implement any one of these.

1. Use locks. Basically you need to ensure that the sequence of operations addChecks -> go addCI() is done atomically. So you would have to acquire a lock in the HTTP handler itself, which is not a very efficient solution imo.
2. Have different files. Just don't forget to clean them up.
    - You can change the file name PRE_COMMIT_CONFIG to have a repo signature. Something like -> `PRE_COMMIT_CONFIG + "_" + repoName`. The script ci.sh would have to copy only this file instead of the entire directory. 
    - You can create a temp directory (which will have an arbitrary name) where you can write PRE_COMMIT_CONFIG, and pass the temp directory name to `ci.sh` so that it would copy the entire directory as it is.